modal/volume.py

```diff
@@ -0,0 +1,106 @@
+# Copyright Modal Labs 2023
+from typing import Optional
+
+from modal_proto import api_pb2
+from modal_utils.async_utils import synchronize_api
+from modal_utils.grpc_utils import retry_transient_errors
+
+from ._resolver import Resolver
+from .object import _Handle, _Provider
+
+
+class _VolumeHandle(_Handle, type_prefix="vo"):
+    """mdmd:hidden Handle to a `Volume` object."""
+
+    async def commit(self):
```

Comment:
commit and reload will only work when run inside a task.

I'm assuming users can call these outside of a task as well somehow? If so, we detect that to raise an appropriate error?

Reply:
> I guess the actual implemention will live in the worker server, so we should be able to just raise an error from another implementation of the extermal rpc call that we put in the actual backend server, which would be the one responding to any non-task client-side calls.

Good idea :+1: 

> Thinking a bit more about it we should also make sure that a user gets an error if they try to commit to a Volume which isn't attached to the currently running task (the task could technically access a "remote" volume)

They will get an error in that case, currently:

![Screenshot from 2023-06-27 17-07-11](https://github.com/modal-labs/modal-client/assets/1300680/fba87e38-d4aa-460e-8722-d186ee853e71)

Should refine that a bit though, to hint that only volumes attached to the current task can be committed/reloaded.

Reply:
https://github.com/modal-labs/modal/pull/6571/commits/2dc5141f95a4ffe51c07392a73cc7ff2dff75e43 - for returning errors when called outside of a function.